### ISSUE TITLE
Setting DJANGO_SETTINGS_MODULE

### ISSUE THREAD

**Issue by sebastian-code** [nedbat/django_coverage_plugin#49]
Right now I have this `setup.cfg` file:
```
## http://pytest.org/latest/customize.html#adding-default-options
[tool:pytest]
DJANGO_SETTINGS_MODULE=config.settings.local
python_files=tests.py test_*.py *_tests.py
addopts=--cov=accountant --cov-config setup.cfg

[coverage:run]
source=accountant/*
omit=*/migrations/*,*/tests/*
# plugins=django_coverage_plugin

[flake8]
max-line-length=80
exclude=.tox,.git,*/migrations/*,*/static/CACHE/*,docs,node_modules,build,dist,*.egg-info
statistics=True

## Run with: pylint --rcfile=setup.cfg somepackage
[pylint]
.
.
.
```
Even when the environment variable `DJANGO_SETTINGS_MODULE` is being set inside the file, the package is not picking it, how is the proper way to do it?

**Comment by PamelaM** [comment]
I think this is really a `pytest` configuration issue and you'll need the pytest-env plugin as discussed in the [stack-overflow response](https://stackoverflow.com/questions/36141024/how-to-pass-environment-variables-to-pytest)

**Comment by PamelaM** [comment]
tl;dr: 
 - install pytest-env plugin
 - modify `setup.cfg` to have an `env=\n` above the `DJANGO_SETTINGS_MODULE` definition and indent that definition.

**Comment by sebastian-code** [comment]
Thanks @PamelaM I tried it but still doesn't work, this time throws a different error thou:
```python
Traceback (most recent call last):
  File "/home/maestro/Workspace/accountant/env/bin/pytest", line 11, in <module>
    sys.exit(main())
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/_pytest/config.py", line 49, in main
    config = _prepareconfig(args, plugins)
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/_pytest/config.py", line 168, in _prepareconfig
    pluginmanager=pluginmanager, args=args)
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 745, in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 339, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 334, in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 613, in execute
    return _wrapped_call(hook_impl.function(*args), self.execute)
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 250, in _wrapped_call
    wrap_controller.send(call_outcome)
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/_pytest/helpconfig.py", line 68, in pytest_cmdline_parse
    config = outcome.get_result()
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 279, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 265, in __init__
    self.result = func()
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 614, in execute
    res = hook_impl.function(*args)
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/_pytest/config.py", line 957, in pytest_cmdline_parse
    self.parse(args)
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/_pytest/config.py", line 1121, in parse
    self._preparse(args, addopts=addopts)
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/_pytest/config.py", line 1092, in _preparse
    args=args, parser=self._parser)
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 745, in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 339, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 334, in <lambda>
    _MultiCall(methods, kwargs, hook.spec_opts).execute()
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 613, in execute
    return _wrapped_call(hook_impl.function(*args), self.execute)
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 254, in _wrapped_call
    return call_outcome.get_result()
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 279, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 265, in __init__
    self.result = func()
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/_pytest/vendored_packages/pluggy.py", line 614, in execute
    res = hook_impl.function(*args)
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/pytest_cov/plugin.py", line 97, in pytest_load_initial_conftests
    plugin = CovPlugin(early_config.known_args_namespace, early_config.pluginmanager)
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/pytest_cov/plugin.py", line 152, in __init__
    self.start(engine.Central)
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/pytest_cov/plugin.py", line 174, in start
    self.cov_controller.start()
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/pytest_cov/engine.py", line 146, in start
    self.cov.erase()
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/coverage/control.py", line 721, in erase
    self._init()
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/coverage/control.py", line 220, in _init
    self.plugins = Plugins.load_plugins(self.config.plugins, self.config, self.debug)
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/coverage/plugin_support.py", line 49, in load_plugins
    coverage_init(plugins, options)
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/django_coverage_plugin/__init__.py", line 11, in coverage_init
    reg.add_file_tracer(DjangoTemplatePlugin())
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/django_coverage_plugin/plugin.py", line 121, in __init__
    check_debug()
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/django_coverage_plugin/plugin.py", line 52, in check_debug
    templates = getattr(settings, 'TEMPLATES', [])
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/django/conf/__init__.py", line 56, in __getattr__
    self._setup(name)
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/django/conf/__init__.py", line 41, in _setup
    self._wrapped = Settings(settings_module)
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/django/conf/__init__.py", line 110, in __init__
    mod = importlib.import_module(self.SETTINGS_MODULE)
  File "/home/maestro/Workspace/accountant/env/lib/python3.5/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 986, in _gcd_import
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 944, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 986, in _gcd_import
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 944, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
  File "<frozen importlib._bootstrap>", line 986, in _gcd_import
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 956, in _find_and_load_unlocked
ImportError: No module named 'config'

```

**Comment by PamelaM** [comment]
Now you're running into regular python path issues where the `config` module isn't in the path or the current working directory.

**Comment by PamelaM** [comment]
Are you able to run a django management command with `DJANGO_SETTINGS_MODULE` set to `config.settings.local`?
e.g.
```
$ DJANGO_SETTINGS_MODULE='config.settings.local' python manage.py migrations list
```

**Comment by sebastian-code** [comment]
Thanks for your time @PamelaM. Yes. Actually, this is how my `manage.py` file looks:
```python
import os
import sys

if __name__ == '__main__':
    os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'config.settings.local')

    from django.core.management import execute_from_command_line

    execute_from_command_line(sys.argv)
```
This file is in the same folder than `setup.cfg`, which is the project's root folder

**Comment by PamelaM** [comment]
Well, I'm honestly stumped.  

A couple more things to try, though:
 - change your module name from `config` to `myconfig` (guess: possible that pytest's config.py is getting in the way of the import search)
 - In your dev environment, edit `/home/maestro/Workspace/accountant/env/lib/python3.5/site-packages/django/conf/__init__.py`, line 110, in __init__ and dump `sys.path` and see that your config module is on the path when running via pytest.  It's possible that the path is getting manipulated somehow.

**Comment by jambonrose** [comment]
@sebastian-code: I'm closing this issue because the issue is nearly a year old. Please let me know if you were not able to track down the error, and I'll re-open the issue and will try to pitch in to help get to the bottom of this.
